### ISSUE TITLE
Crash when doing :undo on Gerrit with Qt 5.10.1

### ISSUE THREAD

**Issue by The-Compiler** [qutebrowser/qutebrowser#3619]
When visiting https://codereview.qt-project.org/#/c/221050/1//ALL,unified and doing `:undo` after closing it with Qt 5.10.1, I get a segfault here:

```
#0  0x00007fffd69b0214 in QtWebEngineCore::WebContentsAdapter::webContents() const (this=0x5555567bfea0) at /tmp/makepkg/qt5-webengine-debug/src/qtwebengine-everywhere-src-5.10.1/src/core/web_contents_adapter.cpp:1188
#1  0x00007fffd69b24ee in QtWebEngineCore::WebContentsAdapter::<lambda()>::operator() (__closure=0x55555aec1040) at /tmp/makepkg/qt5-webengine-debug/src/qtwebengine-everywhere-src-5.10.1/src/core/web_contents_adapter.cpp:605
#2  0x00007fffd69b24ee in QtPrivate::FunctorCall<QtPrivate::IndexesList<>, QtPrivate::List<>, void, QtWebEngineCore::WebContentsAdapter::load(const QWebEngineHttpRequest&)::<lambda()> >::call (arg=<optimized out>, f=...) at /usr/include/qt/QtCore/qobjectdefs_impl.h:130
#3  0x00007fffd69b24ee in QtPrivate::Functor<QtWebEngineCore::WebContentsAdapter::load(const QWebEngineHttpRequest&)::<lambda()>, 0>::call<QtPrivate::List<>, void> (arg=<optimized out>, f=...) at /usr/include/qt/QtCore/qobjectdefs_impl.h:240
#4  0x00007fffd69b24ee in QtPrivate::QFunctorSlotObject<QtWebEngineCore::WebContentsAdapter::load(const QWebEngineHttpRequest&)::<lambda()>, 0, QtPrivate::List<>, void>::impl(int, QtPrivate::QSlotObjectBase *, QObject *, void **, bool *) (which=<optimized out>, this_=0x55555aec1030, r=<optimized out>, a=<optimized out>, ret=<optimized out>) at /usr/include/qt/QtCore/qobjectdefs_impl.h:423
#5  0x00007ffff237e6d4 in QtPrivate::QSlotObjectBase::call(QObject*, void**) (a=0x7fffffffc670, r=<optimized out>, this=<optimized out>) at ../../include/QtCore/../../src/corelib/kernel/qobjectdefs_impl.h:378
#6  0x00007ffff237e6d4 in QSingleShotTimer::timerEvent(QTimerEvent*) (this=0x5555567c6670) at kernel/qtimer.cpp:318
#7  0x00007ffff237213b in QObject::event(QEvent*) (this=0x5555567c6670, e=<optimized out>) at kernel/qobject.cpp:1269
#8  0x00007fffe795507c in QApplicationPrivate::notify_helper(QObject*, QEvent*) (this=this@entry=0x555555ed2b90, receiver=receiver@entry=0x5555567c6670, e=e@entry=0x7fffffffc970) at kernel/qapplication.cpp:3732
#9  0x00007fffe795ca56 in QApplication::notify(QObject*, QEvent*) (this=0x555555f989b0, receiver=0x5555567c6670, e=0x7fffffffc970) at kernel/qapplication.cpp:3491
#10 0x00007fffe5b3e2ef in  () at /usr/lib/python3.6/site-packages/PyQt5/QtWidgets.so
#11 0x00007ffff2340e10 in QCoreApplication::notifyInternal2(QObject*, QEvent*) (receiver=0x5555567c6670, event=event@entry=0x7fffffffc970) at kernel/qcoreapplication.cpp:1050
#12 0x00007ffff239c7a8 in QCoreApplication::sendEvent(QObject*, QEvent*) (event=0x7fffffffc970, receiver=<optimized out>) at ../../include/QtCore/../../src/corelib/kernel/qcoreapplication.h:234
#13 0x00007ffff239c7a8 in QTimerInfoList::activateTimers() (this=0x555556086cd0) at kernel/qtimerinfo_unix.cpp:643
#14 0x00007ffff239cfa2 in timerSourceDispatch(GSource*, GSourceFunc, gpointer) (source=<optimized out>) at kernel/qeventdispatcher_glib.cpp:182
#15 0x00007ffff34f6e38 in g_main_context_dispatch () at /usr/lib/libglib-2.0.so.0
#16 0x00007ffff34f7081 in  () at /usr/lib/libglib-2.0.so.0
#17 0x00007ffff34f710e in g_main_context_iteration () at /usr/lib/libglib-2.0.so.0
#18 0x00007ffff239d361 in QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) (this=0x555556085fe0, flags=...) at kernel/qeventdispatcher_glib.cpp:423
#19 0x00007fffc8be5c62 in QPAEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) (this=0x555556085fe0, flags=...) at qeventdispatcher_glib.cpp:69
#20 0x00007ffff233f44b in QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) (this=this@entry=0x7fffffffcbe0, flags=..., flags@entry=...) at kernel/qeventloop.cpp:212
#21 0x00007ffff2348848 in QCoreApplication::exec() () at kernel/qcoreapplication.cpp:1338
```

Qt 5.10.0 seems to be fine though...

**Comment by SirCmpwn** [comment]
I get the same issue undoing any indiegogo page.

**Comment by The-Compiler** [comment]
I still have no idea why this crash is happening exactly, even after reading the diff between Qt 5.10.0 and 5.10.1. However, I was able to more or less fix it blindly (see d44ff5b) and with that, I can't reproduce it anymore. If you still see it happen somewhere with the latest git master, please let me know!
